src/ripple/app/misc/impl/AmendmentTable.cpp

```diff
@@ -326,31 +362,76 @@ AmendmentTableImpl::AmendmentTableImpl(
     {
-        if (auto s = add(a.first, sl))
-        {
-            JLOG(j_.debug()) << "Amendment " << a.first << " is enabled.";
-
-            if (!a.second.empty())
-                s->name = a.second;
-
-            s->supported = true;
-            s->enabled = true;
+        if (FeatureVotesExist)
+        {  // If the table existed, warn about duplicate config info
+            JLOG(j_.warn()) << "[amendments] section in config file ignored"
+                               " in favor of data in db/wallet.db.";
+            break;
+        }
+        else
+        {  // Otherwise transfer config data into the table
+            persistVote(a.first, a.second, false);  // un-veto
         }
     }
 
+    // Parse enabled amendments from config
     for (auto const& a : parseSection(vetoed))
     {
-        // Unknown amendments are effectively vetoed already
-        if (auto s = get(a.first, sl))
-        {
-            JLOG(j_.info()) << "Amendment " << a.first << " is vetoed.";
-
-            if (!a.second.empty())
-                s->name = a.second;
+        if (FeatureVotesExist)
+        {  // If the table existed, warn about duplicate config info
+            JLOG(j_.warn())
+                << "[veto_amendments] section in config file ignored"
+                   " in favor of data in db/wallet.db.";
+            break;
+        }
+        else
+        {  // Otherwise transfer config data into the table
+            persistVote(a.first, a.second, true);  // veto
+        }
+    }
 
-            s->vetoed = true;
+    // Read amendment votes from wallet.db
+    auto db = db_.checkoutDb();
+    soci::transaction tr(*db);
+    std::string sql =
+        "SELECT AmendmentHash, AmendmentName, Veto FROM FeatureVotes";
+    boost::optional<std::string> amendment_hash;
+    boost::optional<std::string> amendment_name;
+    boost::optional<int> vote;
+    soci::statement st =
+        (db->prepare << sql,
+         soci::into(amendment_hash),
+         soci::into(amendment_name),
+         soci::into(vote));
+    st.execute();
+    while (st.fetch())
+    {
+        uint256 amend_hash = from_hex_text<uint256>(*amendment_hash);
+        if (*vote)  // if veto
+        {
+            // Unknown amendments are effectively vetoed already
+            if (auto s = get(amend_hash, sl))
+            {
+                JLOG(j_.info()) << "Amendment " << amend_hash << " is vetoed.";
+                if (!amendment_name->empty())
+                    s->name = *amendment_name;
+                s->vetoed = true;
+            }
+        }
+        else  // un-veto
+        {
+            if (auto s = add(amend_hash, sl))
+            {
+                JLOG(j_.debug())
+                    << "Amendment " << amend_hash << " is enabled.";
+                if (!amendment_name->empty())
+                    s->name = *amendment_name;
+                s->supported = true;
+                s->enabled = true;
```

Comment:
This appears to perpetuate the bug in the current code when processing the config section. The correct behavior is to not set either `supported` or `enabled`, but only to set `s->vetoed = false;`. For comparison, see https://github.com/ripple/rippled/blob/develop/src/ripple/rpc/handlers/Feature1.cpp#L67-L73

Reply:
Interesting, thanks Ed!  Is there any special action needed in fixing that here as a drive-by?  I.e. does it need its own amendment?

Reply:
I would say "no", but it might be worth some external comms to warn people that it's changing.

My hunch (not backed up by data) is that because rippled votes for new amendments by default, and because this code is so far out doing the wrong thing, there are probably few, if any, operators actually using this config section.